tasks/update_go.py

```diff
@@ -191,8 +191,14 @@ def _update_go_mods(warn: bool, version: str, include_otel_modules: bool, dry_ru
             # to allow them to keep using the modules
             continue
         mod_file = f"./{path}/go.mod"
-        # $ only matches \n, not \r\n, so we need to use \r?$ to make it work on Windows
-        _update_file(warn, mod_file, f"^go {PATTERN_MAJOR_MINOR_BUGFIX}\r?$", f"go {version}", dry_run=dry_run)
+
+        if module.legacy_go_mod_version:
+            minor = _get_major_minor_version(version)
+            # $ only matches \n, not \r\n, so we need to use \r?$ to make it work on Windows
+            _update_file(warn, mod_file, f"^go {PATTERN_MAJOR_MINOR}\r?$", f"go {minor}", dry_run=dry_run)
```

Comment:
nitpick:
```suggestion
            # $ only matches \n, not \r\n, so we need to use \r?$ to make it work on Windows
            _update_file(warn, mod_file, f"^go {PATTERN_MAJOR_MINOR}\r?$", f"go {_get_major_minor_version(version)}", dry_run=dry_run)
```
or rename `minor` in `major_minor`

Reply:
done in 80cf2cbc4dc1e097de243b40e932b4f1030d088a